test/functional/test_framework/test_framework.py

```diff
@@ -479,32 +487,27 @@ def _initialize_chain(self):
             for node in self.nodes:
                 node.wait_for_rpc_connection()
 
-            # For backward compatibility of the python scripts with previous
-            # versions of the cache, set mocktime to Jan 1,
-            # 2014 + (201 * 10 * 60)"""
-            self.mocktime = 1388534400 + (201 * 10 * 60)
-
-            # Create a 200-block-long chain; each of the 4 first nodes
+            # Create a 199-block-long chain; each of the 4 first nodes
```

Comment:
A couple of suggestions that could make this more efficient/clear:

- you could generate all blocks on node0, generating to the other nodes' addresses in turn (you'd need to move the `PRIV_KEYS` constant out of the `get_deterministic_priv_key()` method to access those addresses). This would avoid the `sync_blocks()` calls.
- do any of the tests rely on the immature blocks becoming mature during the test? I'd hope not. If they don't, you could generate those immature blocks to a dummy address, and change this to:

```py
self.nodes[0].generatetoaddress(25, <node0_address>)
self.nodes[0].generatetoaddress(25, <node1_address>)
self.nodes[0].generatetoaddress(25, <node2_address>)
self.nodes[0].generatetoaddress(25, <node3_address>)
self.nodes[0].generatetoaddress(99, <dummy_address>)
```

or:

```py
for i in range(4):
    self.nodes[0].generatetoaddress(25, self.nodes[i].get_deterministic_priv_key().address())
self.nodes[0].generatetoaddress(99, <dummy_address>)
```

Reply:
Done, I measured that it wouldn't make it faster, but at least the code is shorter